kubewarden_test.go

```diff
@@ -0,0 +1,339 @@
+package sdk
+
+import (
+	"testing"
+
+	appsv1 "github.com/kubewarden/k8s-objects/api/apps/v1"
+	batchv1 "github.com/kubewarden/k8s-objects/api/batch/v1"
+	corev1 "github.com/kubewarden/k8s-objects/api/core/v1"
+	"github.com/kubewarden/policy-sdk-go/protocol"
+	"github.com/mailru/easyjson"
+	"github.com/mailru/easyjson/jwriter"
+)
+
+func CreateValidationRequest(object easyjson.Marshaler, kind string) (protocol.ValidationRequest, error) {
+	w := jwriter.Writer{}
+	object.MarshalEasyJSON(&w)
+	value, err := w.BuildBytes()
+	if err != nil {
+		return protocol.ValidationRequest{}, err
+	}
+
+	validationRequest := protocol.ValidationRequest{
+		Settings: easyjson.RawMessage{},
+		Request: protocol.KubernetesAdmissionRequest{
+			Kind: protocol.GroupVersionKind{
+				Kind: kind,
+			},
+			Object: value,
+		},
+	}
+
+	return validationRequest, nil
+}
+
+func CheckIfAutomountServiceAccountTokenIsTrue(t *testing.T, rawResponse []byte) protocol.ValidationResponse {
+	response := protocol.ValidationResponse{}
+	if err := easyjson.Unmarshal(rawResponse, &response); err != nil {
+		t.Fatalf("Error: %v", err)
+	}
+
+	if !response.Accepted {
+		t.Fatalf("Response not accepted")
+	}
+
+	if len(response.MutatedObject.(map[string]interface{})) == 0 {
+		t.Fatalf("Request should be mutated")
+	}
+
+	return response
+}
+
+func TestMutatePodSpecFromRequestWithDeployment(t *testing.T) {
+	deployment := appsv1.Deployment{
+		Spec: &appsv1.DeploymentSpec{
+			Template: &corev1.PodTemplateSpec{
+				Spec: &corev1.PodSpec{
+					AutomountServiceAccountToken: false,
+				},
+			},
+		},
+	}
+
+	validationRequest, err := CreateValidationRequest(deployment, "apps.v1.Deployment")
+	if err != nil {
+		t.Fatalf("Error: %v", err)
+	}
+
+	newPodSpec := corev1.PodSpec{
+		AutomountServiceAccountToken: true,
+	}
+
+	rawResponse, err := MutatePodSpecFromRequest(validationRequest, newPodSpec)
+	if err != nil {
+		t.Fatalf("Error: %v", err)
+	}
+
+	response := CheckIfAutomountServiceAccountTokenIsTrue(t, rawResponse)
+
+	if response.MutatedObject.(map[string]interface{})["spec"].(map[string]interface{})["template"].(map[string]interface{})["spec"].(map[string]interface{})["automountServiceAccountToken"].(bool) != true {
```

Comment:
I don't like this endless list of castings. We know what the mutated object should be (a Deployment in this case).

Can you build a `Deployment` object using the data inside of the `response.MutatedObject` and then write the check?

Note well: these unit tests are running with the official Go compiler, hence you have full reflection power at your disposal (as opposed to the Wasm code)